batch/src/main/scala/uk/gov/ons/addressindex/utils/SqlHelper.scala

```diff
@@ -0,0 +1,27 @@
+package uk.gov.ons.addressindex.utils
+
+import org.apache.spark.sql.DataFrame
+
+/**
+  * Join the Csv files into single DataFrame
+  */
+object SqlHelper {
+
+  def joinCsvs(blpu: DataFrame, lpi: DataFrame, organisation: DataFrame, street: DataFrame,
+               streetDescriptor: DataFrame): DataFrame = {
+    blpu.registerTempTable("blpu")
```

Comment:
Let's improve it. The problem is when you run different tests (when there is more then one test), your tests run in parallel. If you have 2 tests registering a temp table with the same name at the same time, one will overwrite the other. This create race condition.

Add a method to SparkProvider called SparkProvider.createTempTable(dataframe: DataFra,e, tableName: String): String that will accept a dataframe with a table name and return a generated table name.

To avoid problems with concurrency and make this whole thing threadsafe, just append AtomicInteger (a field in SparkProvider) to the name of the table. This way we will be sure that the name of the temp Table is unique across the threads.